figma-shop/src/app/app.component.ts

```diff
@@ -8,10 +8,18 @@ import { Event } from '@angular/router';
 })
 export class AppComponent {
   title = 'figma-shop';
-  componentRemoveFooter = ['ShoppingCartComponent'];
-  isFooterDisplayed = 'show-footer';
+  public showHead: boolean;
 
-  componentAdded(e: Event): void {
-    this.isFooterDisplayed = this.componentRemoveFooter.includes(e.constructor.name) ? 'hide' : 'show';
+  constructor(private router: Router) {
+    this.router.events.forEach((event) => {
+      if (event instanceof NavigationStart) {
+        if (event.url.includes('/admin')) {
+          this.showHead = false;
+        } else {
+          this.showHead = true;
+        }
+      }
+    });
```

Comment:
Please, put this code to a different method, and call it in onInit, try to keep the constructor clearly.